src/tests/end-to-end/tests/details-view/automated-tabstops.test.ts

```diff
@@ -17,86 +20,113 @@ describe('Automated TabStops Results', () => {
+            expect(ruleDetails).toHaveLength(2);
 
-        for (let i = 0; i < 4; i++) {
-            await targetPage.keyPress('Tab');
-            await targetPage.waitForSelectorInShadowRoot(TabStopShadowDomSelectors.svg);
-        }
+            await verifyTargetPageVisualization(1, 1, 2, 0);
+        },
+        longRunningTabStopsTestTimeout,
+    );
 
-        await detailsViewPage.waitForSelector(tabStopsSelectors.automatedChecksResultSection);
-        await detailsViewPage.clickSelector(tabStopsSelectors.failedInstancesExpandButton);
+    test(
+        'Detect and display unreachable elements failures',
+        async () => {
+            await openTabStopsPage('tab-stops/unreachable.html');
 
-        const ruleDetails = await detailsViewPage.getSelectorElements(
-            tabStopsSelectors.failedInstancesContent,
-        );
+            for (let i = 0; i < 5; i++) {
+                await targetPage.keyPress('Tab');
+                await targetPage.waitForSelectorInShadowRoot(TabStopShadowDomSelectors.svg);
+                // Wait longer since this is a keyboard trap
+                await targetPage.waitForTimeout(500);
+            }
 
-        expect(ruleDetails).toHaveLength(1);
-    });
+            await detailsViewPage.waitForSelector(tabStopsSelectors.automatedChecksResultSection);
+            await detailsViewPage.clickSelector(tabStopsSelectors.failedInstancesExpandButton);
 
-    test('Detect and display failures when tabbing is not completed', async () => {
-        await openTabStopsPage('tab-stops/unreachable.html');
+            const ruleDetails = await detailsViewPage.getSelectorElements(
+                tabStopsSelectors.failedInstancesContent,
+            );
 
-        for (let i = 0; i < 2; i++) {
-            await targetPage.keyPress('Tab');
-            await targetPage.waitForSelectorInShadowRoot(TabStopShadowDomSelectors.svg);
-        }
+            expect(ruleDetails).toHaveLength(1);
 
-        // We should just be able to wait for the results section to come up, but there seems to be
-        // a bug in playwright such that it's not recognizing focus on the details view. For now,
-        // toggle the visual helper to get results.
-        await detailsViewPage.setToggleState(tabStopsSelectors.visualHelperToggleButton, false);
+            await verifyTargetPageVisualization(2, 1, 1, 0);
+        },
+        longRunningTabStopsTestTimeout,
+    );
 
-        await detailsViewPage.waitForSelector(tabStopsSelectors.automatedChecksResultSection);
-        await detailsViewPage.clickSelector(tabStopsSelectors.failedInstancesExpandButton);
+    test(
+        'Detect and display failures when tabbing is not completed',
+        async () => {
+            await openTabStopsPage('tab-stops/unreachable.html');
 
-        const ruleDetails = await detailsViewPage.getSelectorElements(
-            tabStopsSelectors.failedInstancesContent,
-        );
+            for (let i = 0; i < 2; i++) {
```

Comment:
nit: Looking at the many cases here, it almost feels like this could be a small named private method as well that can be called in each test.